libraries/AP_Proximity/AP_Proximity_LightWareSF40C.cpp

```diff
@@ -166,280 +197,288 @@ void AP_Proximity_LightWareSF40C::init_sectors()
+                if (sector != last_sector) {
+                    // data for new sector received
+                    // no more data from previous sector will arrive so fill in with shortest distance
+                    if (last_sector != UINT8_MAX) {
+                        if (last_distance_cm < INT16_MAX) {
+                            _angle[last_sector] = last_angle_deg;
+                            _distance[last_sector] = last_distance_cm * 0.01f;
+                            _distance_valid[last_sector] = true;
+                        } else {
+                            _distance_valid[last_sector] = false;
+                        }
+                        // update boundary used for avoidance
+                        update_boundary_for_sector(last_sector, true);
+                    }
+                    // initialise for new sector
+                    last_sector = sector;
+                    last_distance_cm = INT16_MAX;
+                }
+                // use shortest valid distance
+                if ((dist_cm < last_distance_cm) && (dist_cm >= dist_min_cm) && (dist_cm <= dist_max_cm)) {
+                    last_distance_cm = dist_cm;
+                    last_angle_deg = angle_deg;
+                }
             }
-            break;
         }
-
-        default:
-            break;
+        break;
     }
-
-    // mark request as cleared
-    if (success) {
-        _last_request_type = RequestType_None;
+    case MessageID::MOTOR_STATE:
+        if (msg.payload_recv == 1) {
+            sensor_state.motor_state = msg.payload[0];
+        }
+        break;
+    case MessageID::OUTPUT_RATE:
+        if (msg.payload_recv == 1) {
+            sensor_state.output_rate = msg.payload[0];
+        }
+        break;
+
+    // unsupported messages
+    case MessageID::PRODUCT_NAME:
+    case MessageID::HARDWARE_VERSION:
+    case MessageID::FIRMWARE_VERSION:
+    case MessageID::SERIAL_NUMBER:
+    case MessageID::TEXT_MESSAGE:
+    case MessageID::USER_DATA:
+    case MessageID::SAVE_PARAMETERS:
+    case MessageID::STAGE_FIRMWARE:
+    case MessageID::COMMIT_FIRMWARE:
+    case MessageID::INCOMING_VOLTAGE:
+    case MessageID::LASER_FIRING:
+    case MessageID::TEMPERATURE:
+    case MessageID::BAUD_RATE:
+    case MessageID::DISTANCE:
+    case MessageID::MOTOR_VOLTAGE:
+    case MessageID::FORWARD_OFFSET:
+    case MessageID::REVOLUTIONS:
+    case MessageID::ALARM_STATE:
+    case MessageID::ALARM1:
+    case MessageID::ALARM2:
+    case MessageID::ALARM3:
+    case MessageID::ALARM4:
+    case MessageID::ALARM5:
+    case MessageID::ALARM6:
+    case MessageID::ALARM7:
+        break;
     }
+}
 
-    return success;
+// convert buffer to uint32, uint16
+uint32_t AP_Proximity_LightWareSF40C::buff_to_uint32(uint8_t b0, uint8_t b1, uint8_t b2, uint8_t b3) const
+{
+    uint32_t leval = (uint32_t)b0 | (uint32_t)b1 << 8 | (uint32_t)b2 << 16 | (uint32_t)b3 << 24;
+    return le32toh(leval);
```

Comment:
This is actually wrong. The `leval` is already in host endianness - calling this method in BE hosts would make it have the wrong value (I think all our platforms are LE so this is actually a NOP - which is why this is working for you).

Reply:
I don't see how leval could be in host endianness.. this is a buffer received from the sensor .. you're saying that the shifting somehow sorts this out?

I'll remove the le32toh because I suspect you're correct but I don't really understand why actually..